### PR TITLE
[Refactoring] Core tests refactoring

### PR DESCRIPTION
**Pull request by Arsnael**
Just little refactoring moving some test classes from james core to JUnit 5 and some other refactorings

### CODE REVIEW

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 1]
```diff
@@ -102,100 +102,54 @@ void testBadMailAddressString(String mailAddress) {
...
-            assertThat(true).describedAs(a.toString()).isFalse();
-        } catch (AddressException e) {
-            assertThat(true).isTrue();
-        }
-    }
-
-    /**
-     * Test method for {@link MailAddress#MailAddress(javax.mail.internet.InternetAddress)}.
-     */
+    void testGoodMailAddressStringString() {
```
Comment (chibenwa) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: StringString?

Comment (Arsnael) [reply core/src/test/java/org/apache/james/core/MailAddressTest.java]: Didn't change that name... I guess they meant testing the constructor that is accepting two Strings as param, localPart and domain... But I agree we can probably find better

**Comment by Arsnael** [comment]
```
[ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeployment:81 » InvalidServerResponse
```
tests are green... test this please

**Comment by chibenwa** [comment]
`JamesMailSpoolerTest.threadSuicideShouldAckTheItem:122 » ConditionTimeout Cond...`

test this please

**Comment by chibenwa** [comment]
test this please

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 2]
```diff
@@ -37,47 +42,51 @@
...
-            GOOD_QUOTED_LOCAL_PART,
-            "server-dev@james-apache.org",
-            "server-dev@[127.0.0.1]",
-            "server-dev@#123",
-            "server-dev@#123.apache.org",
-            "server.dev@james.apache.org",
-            "\\.server-dev@james.apache.org",
-            "server-dev\\.@james.apache.org",
+    private static Stream<Arguments> goodAddresses() {
+        return Stream.of(
```
Comment (mbaechler) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: well, `map(Arguments::of)` would be cleaner

Comment (Arsnael) [reply core/src/test/java/org/apache/james/core/MailAddressTest.java]: I see... Thanks for the feedback, will do !

**Comment by Arsnael** [comment]
```
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**